src/codegen.c

```diff
@@ -514,8 +516,19 @@ LLVMModuleRef codegen(const CfGraphFile *cfgfile, const FileTypes *ft)
             LLVMSetInitializer(globalptr, LLVMConstNull(t));
     }
 
-    for (CfGraph **g = cfgfile->graphs.ptr; g < End(cfgfile->graphs); g++)
+    for (CfGraph **g = cfgfile->graphs.ptr; g < End(cfgfile->graphs); g++) {
+        if (strcmp((*g)->signature.name, "main") == 0)
+            mainflag = true;
         codegen_function_or_method_def(&st, *g);
+    }
+
+    if (checkmain && mainflag == false) {
+        struct Location l = {
+            .filename = cfgfile->filename,
+            .lineno = 1,
```

Comment:
If you don't want to show a line number at all, you can set `.lineno = 0` here and use an `# Output:` comment in tests.

Reply:
```diff
-missing `main` function to execute the program
+compiler error in file "tests/other_errors/missing_main_function.jou": missing `main` function to execute the program
```
It seems that `Output` doesn't match the error.

Reply:
But the test didn't pass